contracts/lease/src/position/mod.rs

```diff
@@ -7,62 +7,45 @@ use finance::{
 };
 
 use crate::{
-    api::{LeaseCoin, PositionSpecDTO},
+    api::LeaseCoin,
     error::{ContractError, ContractResult},
 };
 
 pub use dto::PositionDTO;
+pub use position_spec::PositionSpec;
 pub use status::{Cause, Liquidation, Status};
 
 mod dto;
+mod position_spec;
 mod status;
 
 #[cfg_attr(test, derive(Debug))]
 pub struct Position<Asset, Lpn> {
     amount: Coin<Asset>,
-    liability: Liability,
-    min_asset: Coin<Lpn>,
-    min_sell_asset: Coin<Lpn>,
+    spec: PositionSpec<Lpn>,
 }
 
 impl<Asset, Lpn> Position<Asset, Lpn>
 where
     Asset: Currency,
     Lpn: Currency,
 {
-    fn new_internal(
-        amount: Coin<Asset>,
-        liability: Liability,
-        min_asset: Coin<Lpn>,
-        min_sell_asset: Coin<Lpn>,
-    ) -> Self {
-        let obj = Self {
-            amount,
-            liability,
-            min_asset,
-            min_sell_asset,
-        };
+    fn new_internal(amount: Coin<Asset>, spec: PositionSpec<Lpn>) -> Self {
+        let obj = Self { amount, spec };
         debug_assert_eq!(Ok(()), obj.invariant_held());
         obj
     }
 
-    pub fn try_from(amount: LeaseCoin, spec: PositionSpecDTO) -> ContractResult<Self> {
-        Ok(Self::new_internal(
-            amount.try_into()?,
-            spec.liability,
-            spec.min_asset.try_into()?,
-            spec.min_sell_asset.try_into()?,
-        ))
+    pub fn try_from(amount: LeaseCoin, spec: PositionSpec<Lpn>) -> ContractResult<Self> {
+        amount
+            .try_into()
+            .map_err(Into::<ContractError>::into)
```

Comment:
The error type is deduced from the function return type. Unless we have any doubts let's skip specifying it.